src/main/java/io/github/intoto/helpers/IntotoHelper.java

```diff
@@ -0,0 +1,158 @@
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SignatureException;
+import java.util.Base64;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+
+/**
+ * Helper class for the intoto-java implementation. This class provides with helper methods to
+ * validate and transform {@link Statement} into their JSON representations.
+ */
+public class IntotoHelper {
+
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static final Validator validator =
+      Validation.buildDefaultValidatorFactory().getValidator();
+
+  /**
+   * Creates a JSON String representation of a DSSE Envelope.
+   *
+   * @param statement the Statement to add to the envelope
+   * @param signer the Signer that will be used to sign the payloads.
+   * @param prettyPrint if true it will pretty print the final Envelope JSON representation
+   * @return a JSON representation for the envelope.
+   * @throws InvalidModelException thrown when the given statement is not valid
+   * @throws JsonProcessingException thrown when there are issues generating the JSON string
+   * @throws NoSuchAlgorithmException thrown when there are issues encrypting the payloads in the
+   *     Envelope
+   * @throws SignatureException thrown when there are issues with the given key in the Signer
+   * @throws InvalidKeyException thrown when there are issues matching the key with the given
+   *     algorithm
+   */
+  public static String produceIntotoEnvelopeAsJson(
+      Statement statement, Signer signer, boolean prettyPrint)
+      throws InvalidModelException, JsonProcessingException, NoSuchAlgorithmException,
+          SignatureException, InvalidKeyException {
+    // Get the Base64 encoded Statement to use as the payload
+    String jsonStatement = validateAndTransformToJson(statement, false);
+    String base64EncodedStatement = Base64.getEncoder().encodeToString(jsonStatement.getBytes());
+
+    IntotoEnvelope envelope = new IntotoEnvelope();
+    // Create the signed payload with the DSSEv1 format and sign it!
+    byte[] signedDsseV1Payload =
+        signer.sign(
+            createPreAuthenticationEncoding(envelope.getPayloadType(), base64EncodedStatement));
+
+    Signature signature = new Signature();
+    signature.setKeyId(signer.getKeyId());
+    // The sig contains the base64 encoded version of the signedDsseV1Payload
+    signature.setSig(Base64.getEncoder().encodeToString(signedDsseV1Payload));
+    // Let's complete the envelope
+    envelope.setPayload(base64EncodedStatement);
+    envelope.setSignatures(List.of(signature));
+    if (prettyPrint) {
+      return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(envelope);
+    }
+    return objectMapper.writeValueAsString(envelope);
```

Comment:
optional: would it make any sense to have the default return value be in JSON Lines format so that it's ready for a Bundle?

Reply:
https://jsonlines.org/ as used by https://github.com/in-toto/attestation/blob/main/spec/bundle.md

basically each JSON object is a _single_ line in the output. So the JSON is rendered without carriage returns, newlines, etc...

Reply:
I guess that should happen in another method that receives a List of Statements? Something like:

```java
String produceIntotoEnvelopeAsJson(
      List<Statement> statement, Signer signer)
      throws InvalidModelException, JsonProcessingException, NoSuchAlgorithmException,
          SignatureException, InvalidKeyException
```

Reply:
Maybe... but people might also have one attestation that they want to put in a bundle.  If it starts with one multi-line JSON it will be a bit messed up.

Reply:
The existing output is not multi-lined. The out put is in a single line.